qutebrowser/config/config.py

```diff
@@ -731,28 +732,49 @@ def set_command(self, win_id, section_=None, option=None, value=None,
             print_ = True
         else:
             with self._handle_config_error():
-                if option.endswith('!') and option != '!' and value is None:
+                if option.endswith('!') and option != '!' and len(values) == 0:
+                    # Handle inversion as special cases of the cycle code path
                     option = option[:-1]
                     val = self.get(section_, option)
-                    layer = 'temp' if temp else 'conf'
                     if isinstance(val, bool):
-                        self.set(layer, section_, option, str(not val).lower())
+                        values = ('false', 'true')
```

Comment:
Presumably because it's a heterogeneous type. I always thought that was “the” distinction between tuples and lists in practice, i.e. with tuples you can't treat each element as being equivalent. (In terms of type/context/whatever)

In the case of set_command it's sort of an exception only because of the way we use the *values.